README.md

```diff
@@ -60,18 +107,36 @@ This is the list of all currently supported blockchains (several coming soon and
-* `runtime.logging.level` - verbosity of the logs. Default - `warning`. Recommended to increase to `debug` when troubleshooting.
-* `runtime.use_exporters` **(work in progress)** - enable other exporters. More on this in [Exporter API](#exporter-api). Default: `false`.
-* `runtime.watchers` - list of enabled watchers (collectors). More on this in [Watchers](#watchers).
+Agent exposes the following configuration parameters to tune its runtime:
+
+Parameter|Details|Env|Type|Default|Required
+-|---|---|---|---|------
+`platform.addr` | Platform GRPC address for publishing. | `MA_PLATFORM` | string | | yes 
+`platform.api_key` | Platform API key for authentication. | `MA_API_KEY` | string | | yes
+`platform.batch_n` | Maximum number of metrics/events to be emitted per publish operation. | `MA_PLATFORM_BATCH_N` | int | 1000 | no 
+`platform.max_publish_interval`| Max interval between publish operations on a healthy state. | `MA_PLATFORM_MAX_PUBLISH_INTERVAL` | duration | 5s | no 
+`platform.transport_timeout` | Publish timeout. | `MA_PLATFORM_TRANSPORT_TIMEOUT` | duration | 10s | no
+`buffer.max_heap_alloc` | Max bytes of allocated heap objects allowed before the agent starts dropping metrics/events. Acts as a limit to prevent unlimited buffering. | `MA_BUFFER_MAX_HEAP_ALLOC` | int | 52428800 | no 
+`buffer.min_buffer_size` | Min number of metrics/events accepted into the buffer regardless if `buffer.max_heap_alloc` has been reached. | `MA_BUFFER_MIN_BUFFER_SIZE` | int | 2500 | no 
+`buffer.ttl` | TTL applied on buffered data. A buffered item is eligible for dropping if its TTL has expired. Use zero to never expire/drop. | `MA_BUFFER_TTL` | duration | 0s | no
+`runtime.logging.outputs` | Any stanard I/O stream (i.e. stdout, stderr), or a full/relative path to a filename (i.e logs/agent.log). | `MA_RUNTIME_LOGGING_OUTPUTS` | comma-separated string | stdout | no
+`runtime.logging.level` | Configures logging level/verbosity. Recommended to increase to `debug` when troubleshooting. | `MA_RUNTIME_LOGGING_LEVEL` | string | warning | no
+`runtime.disable_fingerprint_validation` | Disables host fingerprint validation on startup. |`MA_RUNTIME_DISABLE_FINGERPRINT_VALIDATION` | bool | false | no 
+`runtime.metrics_addr` | Network address to serve agent Prometheus metrics. | `MA_RUNTIME_METRICS_ADDR` | string | 127.0.0.0:9000 | no 
+`runtime.sampling_interval`| Default interval used by watchers polling for data collection. | `MA_RUNTIME_SAMPLING_INTERVAL` | duration | 5s | no 
+`runtime.use_exporters`| **(work in progress)** Enables other exporters. More on this in [Exporter API](#exporter-api). | `MA_RUNTIME_USE_EXPORTERS` | bool | false | no 
+`runtime.watchers`| List of enabled watchers (collectors). More on this in [Watchers](#watchers). | `MA_RUNTIME_WATCHERS` | comma-separated string | | no 
```

Comment:
Yes I agree. I don't like this either but at the same time I understand it's not very helpful if documentation is scattered across places.

Had to change few things in the way agent is handling the configuration when running in a container and thought it'd be a good idea to tidy things up in the current readme to reflect the latest changes and have documented all configuration parameters in one place.

Do you think moving this under `docs/configuration.md` and linking to it would be an option?

Reply:
I'd say a link to `configs/agent.yml` is what I'd prefer since we already got thorough comments there. 

Second solution I'd suggest would be to transfer the documentation for every field to go comments and link to [go docs](https://pkg.go.dev/github.com/Metrika-Inc/agent).

I requested the agent repo to be checked as described [here](https://pkg.go.dev/about#adding-a-package), so it may appear soon if we're not missing anything else.

I don't particularly prefer `docs/configuration.md`, but I would agree it's preferable to having this table in README

Reply:
Addressed in 2b99dbe.